test/e2e/helpers.js

```diff
@@ -545,13 +562,72 @@ async function waitForAccountRendered(driver) {
     '[data-testid="eth-overview__primary-currency"]',
   );
 }
+const WINDOW_TITLES = Object.freeze({
+  ExtensionInFullScreenView: 'MetaMask',
+  TestDApp: 'E2E Test Dapp',
+  Notification: 'MetaMask Notification',
+  ServiceWorkerSettings: 'Inspect with Chrome Developer Tools',
+  InstalledExtensions: 'Extensions',
+});
 
 const logInWithBalanceValidation = async (driver, ganacheServer) => {
   await driver.fill('#password', 'correct horse battery staple');
   await driver.press('#password', driver.Key.ENTER);
   await assertAccountBalanceForDOM(driver, ganacheServer);
 };
 
+async function unlockWallet(driver, walletPassword) {
+  await driver.fill('#password', walletPassword);
+  await driver.press('#password', driver.Key.ENTER);
+}
+
+function roundToXDecimalPlaces(number, decimalPlaces) {
+  return Math.round(number * 10 ** decimalPlaces) / 10 ** decimalPlaces;
+}
+
+function generateRandNumBetween(x, y) {
+  const min = Math.min(x, y);
+  const max = Math.max(x, y);
+  const randomNumber = Math.random() * (max - min) + min;
+
+  return randomNumber;
+}
+
+async function switchToWindow(driver, windowTitle) {
+  const windowHandles = await driver.getAllWindowHandles();
+
+  return await driver.switchToWindowWithTitle(windowTitle, windowHandles);
+}
+
+async function sleepSeconds(sec) {
+  return new Promise((resolve) => setTimeout(resolve, sec * 1000));
+}
+
+async function terminateServiceWorker(driver) {
```

Comment:
we can combine these two helpers together
https://github.com/MetaMask/metamask-extension/blob/develop/test/e2e/helpers.js#LL534C7-L534C27